### PR TITLE
Add parameter "onlyFailing", which will prevent output of successful tests

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix https://github.com/Karumi/Shot/issues/175

### :pushpin: References
* **Issue:** https://github.com/Karumi/Shot/issues/175

### :tophat: What is the goal?
Have additional parameter to only output failing tests, to reduce output on CI

### How is it being implemented?
Optional parameter

### How can it be tested?
- have a failing test
- run it normally, see e.g. 11 successful, 1 failing in comparison
- run it with -PonlyFailing, see only 1 failing in table (headline remains)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -272,6 +272,8 @@ Take into account the ``instrumentationTestTask`` could be different if you use
 
 Shot generates an HTML report you can review at the end of the recording or verification build. However, if you are running Shot in a CI environment which does not support saving the reporting files generated by the build you can verify your tests using this command ``./gradlew debugExecuteScreenshotTests -PprintBase64``. This will change how Shot show comparision errors displaying a command you can copy and paste on your local terminal for every screenshot test failed.
 
+If you want to see only failing tests in output, please add `onlyFailing` as parameter.
```
Comment (pedrovgs) [inline README.md:275]: We should implement this feature as a param of the [extension we created long ago](https://github.com/Karumi/Shot/blob/master/shot/src/main/scala/com/karumi/shot/ShotExtension.scala) as I mentioned in my comments. It is way easier to use.

File: core/src/main/scala/com/karumi/shot/Shot.scala [hunk 2]
```diff
@@ -84,7 +84,8 @@ class Shot(adb: Adb,
                         buildType: String,
                         projectName: String,
                         shouldPrintBase64Error: Boolean,
-                        tolerance: Double): ScreenshotsComparisionResult = {
+                        tolerance: Double,
+                        onlyFailing: Boolean): ScreenshotsComparisionResult = {
```
Comment (pedrovgs) [inline core/src/main/scala/com/karumi/shot/Shot.scala:88]: Can we be more explicit with the variable name? Something like ``showOnlyFailingTestsInReports`` or something similar?

File: core/src/main/scala/com/karumi/shot/reports/ExecutionReporter.scala [hunk 3]
```diff
@@ -141,20 +144,27 @@ class ExecutionReporter {
...
       .mkString("\n")
   }
 
   private def generateScreenshotsTableBody(
-      comparision: ScreenshotsComparisionResult): String = {
+      comparision: ScreenshotsComparisionResult,
+      onlyFailing: Boolean): String = {
     getSortedByResultScreenshots(comparision)
       .map {
+        // case (_, None) => "" // we do not show it
```
Comment (pedrovgs) [inline core/src/main/scala/com/karumi/shot/reports/ExecutionReporter.scala:167]: Can we remove this comment if this case is not needed anymore?

**Review by pedrovgs — CHANGES_REQUESTED** [review]
@tobiasKaminsky the code is ok apart from the detail we should use an extension for the configurable param instead of using a parameter. The build is failing because you have to format the code you modified. You can use ``./gradlew scalafmtAll`` :) I'm also wondering if once you've created the extension param, you could modify one of the consumers to use it. This would work as a smoke test for us 😃  Thank you so much for your time working on this feature. I really appreciate it!

File: README.md (hunk 1 above)
Comment (tobiasKaminsky) [reply README.md:275]: I currently fail to understand how this ShotExtension can be then configured.
It seems for me like this are "just" constants, which cannot be changed during runtime?

File: core/src/main/scala/com/karumi/shot/Shot.scala (hunk 2 above)
Comment (tobiasKaminsky) [reply core/src/main/scala/com/karumi/shot/Shot.scala:88]: Sure, I'll change it

**Comment by tobiasKaminsky** [comment]
>  You can use `./gradlew scalafmtAll`

This heavily changed imports and thus nothing worked, so I do not dare to check this in ;-)



> I'm also wondering if once you've created the extension param, you could modify one of the consumers to use it.

I added it to shot-consumer :+1:

**Comment by pedrovgs** [comment]
@tobiasKaminsky I've formatted the code for you using the command I mentioned and updated one of the consumers to use the extension param you created. The PR is ready to be merged! Thank you so much for your time and contribution in this PR.

**Review by pedrovgs — APPROVED** [review]

**Pull request merged**